crates/proof-of-sql/src/base/commitment/vec_commitment_ext.rs

```diff
@@ -440,37 +427,38 @@ mod tests {
     fn we_can_sub_commitment_collections() {
         let column_a = [12i64, 34, 56, 78, 90];
         let column_b = ["Lorem", "ipsum", "dolor", "sit", "amet"].map(String::from);
+        let front_emptied_column_a = [0i64, 0, 0, 78, 90];
+        let front_emptied_column_b = ["", "", "", "sit", "amet"].map(String::from);
 
         let columns = vec![
-            OwnedColumn::<Curve25519Scalar>::BigInt(column_a[..3].to_vec()),
+            OwnedColumn::<TestScalar>::BigInt(column_a[..3].to_vec()),
             OwnedColumn::VarChar(column_b[..3].to_vec()),
         ];
 
-        let commitments_a = Vec::<RistrettoPoint>::from_columns_with_offset(&columns, 0, &());
+        let commitments_a = Vec::<NaiveCommitment>::from_columns_with_offset(&columns, 0, &());
 
         let full_columns = vec![
-            OwnedColumn::<Curve25519Scalar>::BigInt(column_a.to_vec()),
+            OwnedColumn::<TestScalar>::BigInt(column_a.to_vec()),
             OwnedColumn::VarChar(column_b.to_vec()),
         ];
 
-        let commitments_b = Vec::<RistrettoPoint>::from_columns_with_offset(&full_columns, 0, &());
+        let commitments_b = Vec::<NaiveCommitment>::from_columns_with_offset(&full_columns, 0, &());
 
         let commitments = commitments_b.try_sub(commitments_a).unwrap();
 
-        let mut expected_commitments = vec![CompressedRistretto::default(); 2];
-        compute_curve25519_commitments(
-            &mut expected_commitments,
-            &[
-                Sequence::from(&column_a[3..]),
-                Sequence::from(&column_b.map(Curve25519Scalar::from).map(<[u64; 4]>::from)[3..]),
-            ],
-            3,
-        );
-        let expected_commitments: Vec<_> = expected_commitments
-            .iter()
-            .map(|c| c.decompress().unwrap())
-            .collect();
+        let committable_columns = [
+            CommittableColumn::BigInt(&front_emptied_column_a),
+            CommittableColumn::VarChar(
+                front_emptied_column_b
+                    .iter()
+                    .map(Into::<TestScalar>::into)
+                    .map(Into::<[u64; 4]>::into)
+                    .collect(),
+            ),
+        ];
 
+        let expected_commitments =
+            NaiveCommitment::compute_commitments(&committable_columns, 0, &());
```

Comment:
Something like this is better.
```suggestion
            CommittableColumn::BigInt(&column_a[3..]),
            CommittableColumn::VarChar(
                column_b[3..]
                    .iter()
                    .map(Into::<TestScalar>::into)
                    .map(Into::<[u64; 4]>::into)
                    .collect(),
            ),
        ];

        let expected_commitments =
            NaiveCommitment::compute_commitments(&committable_columns, 0\3, &());
```

Reply:
Modified it